lib/logstash/filters/jdbc/db_object.rb

```diff
@@ -0,0 +1,148 @@
+require_relative "validatable"
+require_relative "column"
+
+module LogStash module Filters module Jdbc
+
+  TEMP_TABLE_PREFIX = "temp_".freeze
+  DB_OBJECT_TYPES = ["table".freeze, "index".freeze].freeze
+
+  class DbObject < Validatable
+    #   {type => "table", name => "servers", columns => [["ip", "text"], ["name", "text"], ["location", "text"]]},
+    #   {type => "index", name => "servers_idx", table => "servers", columns => ["ip"]}
+
+    attr_reader :type, :name, :table, :columns, :builder
+
+    def build(db)
+      return unless valid?
+      if index?
+        IndexBuilder.new(@name, @columns, @table).build(db)
+      else
+        TableBuilder.new(@name, @columns).build(db)
+      end
+    end
+
+    class TableBuilder
+      attr_reader :name, :columns
+      def initialize(name, columns)
+        @name, @columns = name, columns
+      end
+      def build(db)
+        schema_gen = db.create_table_generator()
+        @columns.each {|col| schema_gen.column(col.name, col.datatype)}
+        options = {:generator => schema_gen}
+        db.create_table(@name, options)
```

Comment:
This could generate errors, if the user doesn't have the permission to create a table on the database, we need to trap that.

Reply:
OK, technically true, but this is the local DB. If it is a memory based Derby that we initialise then chances are v limited. However if the user provides their own, say Postgres, DB then yes, likely. +1